compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -39,10 +39,37 @@ use rustc_hir::def_id::DefId;
 use rustc_hir::def_id::LocalDefId;
 use rustc_hir::intravisit::{self, NestedVisitorMap, Visitor};
 use rustc_infer::infer::UpvarRegion;
-use rustc_middle::hir::place::{PlaceBase, PlaceWithHirId};
+use rustc_middle::hir::place::{Place, PlaceBase, PlaceWithHirId, ProjectionKind};
 use rustc_middle::ty::{self, Ty, TyCtxt, UpvarSubsts};
+use rustc_span::sym;
 use rustc_span::{Span, Symbol};
-use std::collections::hash_map::Entry;
+
+use std::env;
+
+macro_rules! log_capture_analysis {
+    ($fcx:expr, $closure_def_id:expr, $fmt:literal) => {
+        if $fcx.should_log_capture_analysis($closure_def_id) {
+            debug!("For closure={:?}: {}", $closure_def_id, $fmt);
+        }
+    };
+
+    ($fcx:expr, $closure_def_id:expr, $fmt:literal, $($args:expr),*) => {
+        if $fcx.should_log_capture_analysis($closure_def_id) {
+            debug!(concat!("For closure={:?}: ", $fmt), $closure_def_id, $($args),*);
+        }
+    };
+}
```

Comment:
Should we keep this macro? Maybe let's just remain consistent and move everything to aprinting method?

Reply:
Updated the last commit